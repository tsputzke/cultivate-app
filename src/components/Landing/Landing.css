* {
  box-sizing: border-box;
}

.landing-title {
  text-align: center;
}

p {
  margin-bottom: 50px;
}

.landing {
  margin: auto;
  padding-top: 100px;
  line-height: 1.5;
  width: 80%;
}

#landing-section {
  padding: 15px 30px;
  text-align: center;
  margin: auto;
  border: 1px solid black;
  width: 250px;
  height: 250px;
}

#user-button {
  border: 3px solid red;
}