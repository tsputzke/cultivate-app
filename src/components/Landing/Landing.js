import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import AuthApiService from '../../services/auth-api-service'
import TokenService from '../../services/token-service'
import UserContext from '../../context/user-context'

export default class Landing extends Component {
  static contextType = UserContext
  state = { error: null }
  
  handleLogin = e => {
    e.preventDefault();

    this.setState({ error: null })
    const { user_name, password } = e.target

    const user = {
      user_name: user_name.value.trim(),
      password: password.value.trim(),
    }
    
    AuthApiService.postLogin(user)
      .then(res => {
        user_name.value = ''
        password.value = ''
        TokenService.saveAuthToken(res.authToken)
        window.sessionStorage.setItem('user_id', res.user_id)
        window.sessionStorage.setItem('user_name', user.user_name)
        window.sessionStorage.setItem('isLoggedIn', 'true')
        this.context.toggleState()
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
    }
  render() {
    const { error } = this.state
    return (
      <div className='landing'>
        <h1 className='landing-title center-align title-style'>
          Cultivate  
        </h1>
        <p className='landing-description'>
          Grow smarter. Track your grow-room data with Cultivate. <span><Link to='/about'><u>Learn more!</u></Link></span>
        </p>
        <section className="login">
          <form
            className="login-form"
            onSubmit={this.handleLogin}
          >
            <fieldset>
              <legend className="strong login-legend">User Login:</legend>
              <div>
                <label htmlFor="user_name"> Username: </label>
                <input type="text" name="user_name" required />
              </div>
              <div>
                <label htmlFor="password">Password: </label>
                <input type="password" name="password" required />
              </div>
              <button className="login-button" type="submit">Login</button>
              <section className="login-info">
                <p><Link to='/registration'><u>New User?</u></Link></p>
                <hr />
                <p><strong>To login as a test user:</strong> <br /> <u>Username:</u> TestUser <br /> <u>Password:</u> testuser1!</p>
              </section>
            </fieldset>
          </form>
          <div role='alert'>
            {error && <p className='error'>{error}</p>}
          </div>
        </section>
      </div>
    )
  }
}