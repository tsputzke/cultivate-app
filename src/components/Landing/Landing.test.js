import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Landing from './Landing'

configure({adapter: new Adapter()});  

describe(`Landing component`, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <Landing />
      </BrowserRouter>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders Landing by default', () => {
    const wrapper = shallow(<Landing />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})