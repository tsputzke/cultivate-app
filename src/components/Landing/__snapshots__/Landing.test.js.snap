// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component renders Landing by default 1`] = `
<div
  className="landing"
>
  <h1
    className="landing-title center-align title-style"
  >
    Cultivate
  </h1>
  <p
    className="landing-description"
  >
    Grow smarter. Track your grow-room data with Cultivate. To learn more, 
    <span>
      <Link
        to="/about"
      >
        <u>
          click here
        </u>
      </Link>
    </span>
    .
  </p>
  <section
    className="login"
  >
    <form
      className="login-form"
      onSubmit={[Function]}
    >
      <fieldset>
        <legend
          className="strong login-legend"
        >
          User Login:
        </legend>
        <div>
          <label
            htmlFor="user_name"
          >
             Username: 
          </label>
          <input
            name="user_name"
            required={true}
            type="text"
          />
        </div>
        <div>
          <label
            htmlFor="password"
          >
            Password: 
          </label>
          <input
            name="password"
            required={true}
            type="password"
          />
        </div>
        <button
          className="login-button"
          type="submit"
        >
          Login
        </button>
        <section
          className="login-info"
        >
          <p>
            New user? 
            <span>
              <Link
                to="/registration"
              >
                <u>
                  register here
                </u>
              </Link>
            </span>
          </p>
          <hr />
          <p>
            <strong>
              Login as a test user:
            </strong>
             
            <br />
             
            <u>
              Username:
            </u>
             TestUser 
            <br />
             
            <u>
              Password:
            </u>
             testuser1!
          </p>
        </section>
      </fieldset>
    </form>
    <div
      role="alert"
    />
  </section>
</div>
`;
